llvm/lib/MCCAS/MCCASObjectV1.cpp

```diff
@@ -2059,24 +2156,38 @@ Error MCCASBuilder::buildFragments() {
     // Start subsection for first Atom.
     startAtom(Sec.getFragmentList().front().getAtom());
 
+    SmallVector<char, 0> Addends;
+    ArrayRef<MachO::any_relocation_info> RelocationBuffer;
     MCDataFragmentMerger Merger(*this, &Sec);
+    uint64_t RelocationBufferIndex = 0;
     for (const MCFragment &F : Sec) {
+      auto Relocs = RelMap.find(&F);
       if (RelocLocation == Atom) {
-        auto Relocs = RelMap.find(&F);
-        if (Relocs != RelMap.end())
+        if (Relocs != RelMap.end()) {
           AtomRelocs.append(Relocs->second.begin(), Relocs->second.end());
-        createAddendVector(AtomAddends, ObjectWriter.getAddends()[&F], &F,
-                           Layout);
-      } else
-        createAddendVector(SectionAddends, ObjectWriter.getAddends()[&F], &F,
-                           Layout);
-
+          // Reset RelocationBufferIndex if Relocations exist per Fragment.
+          RelocationBufferIndex = 0;
+        }
+      }
+      // Relocations can be stored in the section or be in the RelMap, which
+      // means they will be stored with the Atom, but currently can be accessed
```

Comment:
Here's an attempt to untangle this comment:
Relocations can be stored in the Section or in the Atom. Atom relocation are entered into RelMap.

I don't know what "currently" means here.

Reply:
The RelMap is per fragment, but regardless of whether the relocations are in the RelMap or in the SectionRelocs, they will be stored in a sectionref or an AtomRef, not in a FragmentRef, that is why I said currently, the step of storing the relocations in the CAS happens later, not at this point in processing

Reply:
Which is why the comment you have is a little backwards, what I am trying to say is, the relocations right now are accessible via the RelMap or the SectionRelocs, if they are with the SectionRelocs, they will go in the SectionRef, if they are in the RelMap, they will go in the AtomRef, but the RelMap is per fragment,  not per Atom.